core/services/relay/evm/binding/lookup.go

```diff
@@ -0,0 +1,79 @@
+package binding
+
+import (
+	"sync"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+)
+
+type readValues struct {
+	address  string
+	contract string
+	method   string
+}
+
+type lookup struct {
+	mu              sync.RWMutex
+	contractMethods map[string][]string
```

Comment:
These fields need better naming and comments, I don't think that anyone that isn't actively developing this will be able to understand whats going on. I wrote a quick suggestion, maybe its too verbose 

```suggestion
type readValues struct {
	address  string
	contract string
	method   string
}

// readsNames is a slice of contract method and event names
type readsNames []string
type lookup struct {
	mu sync.RWMutex
	// key is contractName
	contractsReadsNames map[string]readsNames
	// key is read identifier (contractName + address + readName)
	readIdentifierParser map[string]readValues
}
```